clusterDeployer.py

```diff
@@ -479,55 +471,61 @@ def cb() -> None:
                 worker.set_password()
 
         # Make sure any submitted tasks have completed.
-        for p in futures:
-            p.result()
-
-    def _rename_workers(self, infra_env: str) -> None:
-        logger.info("Waiting for connectivity to all workers")
-        hosts = []
-        workers = []
-        for bm in self._all_hosts:
-            for k8s_node in bm.k8s_worker_nodes:
-                rh = host.RemoteHost(k8s_node.ip())
-                rh.ssh_connect("core")
-                hosts.append(rh)
-                workers.append(k8s_node)
+        for pair in nodes_with_futures:
+            node, _ = pair
+            node.future.result()
+
+    def _start_node(self, infra_env: str, node: ClusterNode, master: bool, executor: ThreadPoolExecutor) -> bool:
+        image = os.path.join(os.path.dirname(node.config.image_path), f"{infra_env}.iso")
+        node.start(image, executor)
+        node.wait_for_boot(self._cc.full_ip_range)
+
+        if not master and not self._rename_worker(infra_env, node):
+            return False
+
+        return self._wait_known(node)
+
+    def _rename_worker(self, infra_env: str, node: ClusterNode) -> bool:
+        logger.info(f"Waiting for connectivity to worker {node.config.name}")
+        rh = host.RemoteHost(node.ip())
+        rh.ssh_connect("core")
 
         ip_range = self._cc.full_ip_range
-        logger.info(f"Connectivity established to all workers; checking that they have an IP in range: {ip_range}")
+        logger.info(f"Connectivity established to worker {node.config.name} checking that it has an IP in range: {ip_range}")
 
-        any_worker_bad = False
-        for w, h in zip(workers, hosts):
-            if not common.any_address_in_range(h, ip_range):
-                logger.error(f"Worker {w.config.name} doesn't have an IP in range {ip_range}.")
-                any_worker_bad = True
+        if not common.any_address_in_range(rh, ip_range):
+            logger.error(f"Worker {node.config.name} doesn't have an IP in range {ip_range}.")
+            return False
 
-        if any_worker_bad:
-            sys.exit(-1)
+        for try_count in range(30):
+            info = self._ai.get_ai_host_id_by_ip(infra_env, node.ip())
+            if info is not None:
+                self._ai.update_host(info.id, {"name": node.config.name})
+                logger.info(f"Renamed {node.config.name}")
+                return True
```

Comment:
what if we fail 30x

Reply:
Then the whole `_start_node` will fail, so it will retry the installation again.